src/Tutor.Web/Controllers/Domain/KCController.cs

```diff
@@ -68,13 +68,28 @@ public ActionResult<AssessmentEventDto> GetSuitableAssessmentEvent([FromBody] As
             if (result.IsSuccess) return Ok(_mapper.Map<AssessmentEventDto>(result.Value));
             return NotFound(result.Errors);
         }
-        
+
         [HttpGet("knowledge-components/mastery/{knowledgeComponentId:int}")]
         public ActionResult<KnowledgeComponentMasteryDto> GetKnowledgeComponentMastery(int knowledgeComponentId)
         {
             var result = _kcService.GetKnowledgeComponentMastery(int.Parse(User.Claims.First(i => i.Type == "id").Value), knowledgeComponentId);
             if (result.IsSuccess) return Ok(_mapper.Map<KnowledgeComponentMasteryDto>(result.Value));
             return NotFound(result.Errors);
         }
+
+        [HttpPost("knowledge-components/{knowledgeComponentId:int}/session/launch")]
+        public ActionResult LaunchSession(int knowledgeComponentId)
+        {
+            _kcService.LaunchSession(int.Parse(User.Claims.First(i => i.Type == "id").Value), knowledgeComponentId);
```

Comment:
Can we extract `int.Parse(User.Claims.First(i => i.Type == "id").Value` into a separate method?

Reply:
Extracted in https://github.com/Clean-CaDET/tutor/pull/30/commits/66be496dd6889d39e8527ac4462d8659e3c9d753. I wasn't quite sure where to put the method since it will be needed in most controllers, so I made it an extension method for ClaimsPrincipal and I put it in Infrastructure.Security.Authorization.JWT for now.